utils_test.py

```diff
@@ -481,7 +481,24 @@ def test_get_time_in_millisecs_with_complicated_time(self):
             datetime.datetime.fromtimestamp(python_utils.divide(msecs, 1000.0)))
 
     def test_get_current_appengine_environment(self):
+        saved_appengine_runtime = (
+            os.environ['APPENGINE_RUNTIME'] if 'APPENGINE_RUNTIME' in os.environ
+            else None)
+        saved_server_software = (
+            os.environ['SERVER_SOFTWARE'] if 'SERVER_SOFTWARE' in os.environ
+            else None)
+
         os.environ['APPENGINE_RUNTIME'] = 'True'
-        self.assertTrue(utils.is_appengine_simulated_environment())
+        self.assertTrue(utils.is_local_server_environment())
         os.environ['SERVER_SOFTWARE'] = 'Google App Engine/'
         self.assertTrue(utils.is_appengine_cloud_environment())
+
+        if saved_appengine_runtime:
```

Comment:
Technically, here and below, it should be "if XXX is not None".

Reply:
Done, thanks for the catch!